js/modules/venn.src.js

```diff
@@ -0,0 +1,742 @@
+            combinations.push(set + ',' + set2);
+        });
+        return combinations;
+    }, []).forEach(function (combination) {
+        if (!mapOfIdToRelation[combination]) {
+            var obj = {
+                sets: combination.split(','),
+                value: 0
+            };
+            mapOfIdToRelation[combination] = obj;
+        }
+    });
+
+    // Transform map into array.
+    return objectValues(mapOfIdToRelation);
+};
+
+/**
+ * getScale - Calculates the proper scale to fit the cloud inside the plotting
+ *            area.
+ *
+ * NOTE: copied from wordcloud.
+ *
+ * @param  {number} targetWidth  Width of target area.
+ * @param  {number} targetHeight Height of target area.
+ * @param  {object} field The playing field.
+ * @param  {Series} series Series object.
+ * @return {number} Returns the value to scale the playing field up to the size
+ *     of the target area.
+ */
+var getScale = function getScale(targetWidth, targetHeight, field) {
+    var height = Math.max(Math.abs(field.top), Math.abs(field.bottom)) * 2,
+        width = Math.max(Math.abs(field.left), Math.abs(field.right)) * 2,
+        scaleX = width > 0 ? 1 / width * targetWidth : 1,
+        scaleY = height > 0 ? 1 / height * targetHeight : 1;
+    return Math.min(scaleX, scaleY);
+};
+
+/**
+ * updateFieldBoundaries - If a circle is outside a give field, then the
+ * boundaries of the field is adjusted accordingly. Modifies the field object
+ * which is passed as the first parameter.
+ *
+ * NOTE: Copied from wordcloud, can probably be unified.
+ *
+ * @param  {object} field The bounding box of a playing field.
+ * @param  {object} placement The bounding box for a placed point.
+ * @return {object} Returns a modified field object.
+ */
+var updateFieldBoundaries = function updateFieldBoundaries(field, circle) {
+    var left = circle.x - circle.r,
+        right = circle.x + circle.r,
+        bottom = circle.y + circle.r,
+        top = circle.y - circle.r;
+
+    // TODO improve type checking.
+    if (!isNumber(field.left) || field.left > left) {
+        field.left = left;
+    }
+    if (!isNumber(field.right) || field.right < right) {
+        field.right = right;
+    }
+    if (!isNumber(field.top) || field.top > top) {
+        field.top = top;
+    }
+    if (!isNumber(field.bottom) || field.bottom < bottom) {
+        field.bottom = bottom;
+    }
+    return field;
+};
+
+var getDlOptions = function getDlOptions(params) {
+    // Set options to new object to avoid problems with scope
+    var optionsPoint = (
+            isObject(params.optionsPoint) ?
+            params.optionsPoint.dataLabels :
+            {}
+        ),
+        optionsSeries = (
+            isObject(params.optionsSeries) ?
+            params.optionsSeries.dataLabels :
+            {}
+        ),
+        options = merge({
+            style: {}
+        }, optionsSeries, optionsPoint);
+    return options;
+};
+
+var vennOptions = {
```

Comment:
Where are the doclets? Test with `gulp jsdoc`.

Reply:
Doclets are added to the branch, and api of venn series type has been reviewed.